src/handlers/pull-reviewer.ts

```diff
@@ -275,16 +240,16 @@ export class PullReviewer {
     });
 
     if (closingIssues.length === 0) {
-      const linkedViaBodyHash = pull_request.body?.match(/#(\d+)/g);
+      const linkedViaBodyHash = pull_request.body?.match(/[Rr]esolves\s+#(\d+)/);
 
       if (linkedViaBodyHash?.length) {
-        issueNumber = Number(linkedViaBodyHash[0].replace("#", ""));
+        issueNumber = Number(linkedViaBodyHash[1]);
       } else {
         await this.convertPullToDraft(context.payload.pull_request.node_id, context.octokit);
-        throw context.logger.error("You need to link an issue and after that convert the PR to ready for review");
+        throw this.context.logger.error("You need to link an issue and after that convert the PR to ready for review");
       }
```

Comment:
It's pointless to check the body because if you write `Resolves #123` it will be in linked issues. If there's no linked issues then convert to draft and warn the user